### PR TITLE
Web - Attachments Preview_3901

### PR DESCRIPTION
**Pull request by MTN718**
### Details

Fix attachment view zoom issue

### Fixed Issues

$ https://github.com/Expensify/App/issues/3901

### Tests

1. On chat screen, send attachment image.
2. Open attachment view
3. Try click zoom and drag

### Tested On
- [x] Web
- [ ] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

Web

#### Desktop

Mac_Desktop

#### iOS

Iphone

#### Android

Android

### CODE REVIEW

**Comment by MTN718** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by roryabraham** [comment]
@MTN718 minor detail to keep in mind for future PRs: including the full url of the linked issue in the PR description (instead of just the issue number) helps link the PR to the associated issue, and helps our bots choose a reviewer for the PR.

**Review by roryabraham — COMMENTED** [review]
Overall the approach you've taken in this PR looks good, but we need to get the lint errors resolved before I test + merge. Thanks

**Comment by MTN718** [comment]
Ok Sorry first time to make PR. I will improve for next PR

**Comment by roryabraham** [comment]
No worries, this is totally normal for a first-time contributor. 😄  It always takes a little while to get used to a new repo, and we have pretty strict code style guidelines, which are outlined [here](https://github.com/Expensify/App/blob/main/STYLE.md).

**Comment by MTN718** [comment]
So now I need to wait until you review correct?

**Comment by robertjchen** [comment]
@MTN718 No need to wait, please feel free to fix the lint errors first and we can review after 👍

**Comment by MTN718** [comment]
link error fixed

File: src/styles/styles.js
Comment (robertjchen) [inline src/styles/styles.js:2179]: I would probably move this `const` into https://github.com/Expensify/App/blob/main/src/CONST.js, similar to what we do for another constant in this file: https://github.com/Expensify/App/blob/e171217e4d9cf089d0d7f4053b92c42107fe312e/src/styles/styles.js#L2357

**Review by robertjchen — CHANGES_REQUESTED** [review]

File: src/styles/styles.js
Comment (MTN718) [reply src/styles/styles.js:2179]: Ok.

Comment (MTN718) [reply src/styles/styles.js]: Fixed

**Review by robertjchen — DISMISSED** [review]
Looks good to me, I'll let @roryabraham do the honors! 👍

**Review by roryabraham — CHANGES_REQUESTED** [review]
Overall I found two problems with this implementation:

1. I find the zoom factor to be a bit too extreme, especially on smaller devices. I think we need a better heuristic to scale the `MIN_ZOOM_SIZE` in accordance with the device height/width.
1. Zoom is not centered on the user's click. For example, clicking on the right side of the image does not zoom in on the right side of the image:

https://user-images.githubusercontent.com/47436092/130484870-38da7d04-55e3-4d38-8c0f-ae439d853bfa.mov

**Comment by MTN718** [comment]
2. Zoom is not centered on the user's click. For example, clicking on the right side of the image does not zoom in on the right side of the image:
So it should be show always center of the image when zoom?

**Comment by MTN718** [comment]
@roryabraham  fixed issues. Please review

File: src/components/ImageView/index.js
Comment (roryabraham) [inline src/components/ImageView/index.js:67]: I think this `if/else` block can be clarified by extracting some descriptive local variables and adding some comments:

```javascript
const isScreenWiderThanImage = (this.state.screenWidth / width) > 1;
const isScreenTallerThanImage = (this.state.screenHeigh / height) > 1;
if ((aspect > 1 && isScreenWiderThanImage) || (aspect <= 1 && isScreenTallerThanImage)) {
    // Descriptive comment to describe how x1, x2, y1, y2 are set
} else if (aspect > 1 && !isScreenWiderThanImage) {
    // Descriptive comment
} else if (aspect <= 1 && !isScreenTallerThanImage) {
    // Descriptive comment
}
```

**Review by roryabraham — CHANGES_REQUESTED** [review]
Overall this seems to work pretty well, but I think we need to clean up the code a bit and leave more descriptive variable names + comments to explain how this is all working.

File: src/components/ImageView/index.js
Comment (MTN718) [reply src/components/ImageView/index.js:32]: when open attach view, image in center, at that time, left image x point. and rx2 is right of image x point

File: src/components/ImageView/index.js
Comment (MTN718) [reply src/components/ImageView/index.js:43]: I will check that.

File: src/components/ImageView/index.js
Comment (MTN718) [reply src/components/ImageView/index.js:61]: Ok.

File: src/components/ImageView/index.js
Comment (MTN718) [reply src/components/ImageView/index.js:67]: Ok.

**Comment by MTN718** [comment]
Update Code following your feedback @roryabraham

**Comment by MTN718** [comment]
@roryabraham any feedback?

**Review by roryabraham — APPROVED** [review]
Thanks, this is looking much better! The last round of changes makes the code much more approachable, so thanks for making the effort.

**Comment by MTN718** [comment]
Thanks. Will keep this styling.

**Comment by roryabraham** [comment]
Great job getting your first Expensify/App pull request over the finish line! I'll merge it assuming all the tests pass.

I know there's a lot of information in our contributing guidelines, so some points to take note of:

1. Once your PR is merged, you can be hired for another issue. Once you've completed a few issues, we may start hiring you for more than one issue at a time.
1. Once your PR is deployed to our staging servers, it will undergo quality assurance (QA) testing. If we find that this doesn't work as expected or causes a regression, you'll be responsible for fixing it. Typically we would revert this PR and give you another chance to create a similar PR without causing a regression. (I don't imagine this will happen with this PR, but it's something to be aware of)
1. Once your PR is deployed to _production_, we start a 7-day timer. After it has been on production for 7 days without causing any regressions, then we pay out the Upwork job.

So it might take a while before you're paid for your work, but we typically post multiple new jobs every day, so there's plenty of opportunity. I hope you've had a positive experience contributing to this repo!

**Comment by MTN718** [comment]
Ok No problem. I submitted one more proposal for another issue seems you
assigned that issue. can you check my proposal?

>

**Review by robertjchen — APPROVED** [review]

**Pull request merged**
